resources/views/blade/icon.blade.php

```diff
@@ -0,0 +1,10 @@
+{{-- IMPORTANT!!! Make sure there is no newline at the end of this file, or it will break the loaders for the tables --}}
+
+@props([
+    'type' => '',
+    'class' => false,
+    'style' => false,
+    'id' => false,
+    'title' => false,
+])
+<i {{ $attributes->merge(['class' => Icon::icon($type).' '.$class]) }} {{ isset($style) ? $attributes->merge(['style' => $style]): '' }}  {{ isset($title) ? $attributes->merge(['title' => $title]): '' }}  aria-hidden="true"></i>
```

Comment:
I think, since the only place the 'Icon helper' is used is right here, that you might be able to directly paste it in above this `<i>` tag, in a `@php` block.

Then the thing that spits out the icons is right next to the place we use the icons, right? And you can get rid of some other extra little bits here and there.

Reply:
There are some other places where we're not directly generating an icon itself though (some transformers, etc) so it seemed better to keep that logic out of the blade component.

Reply:
In fact, you can just make it into a simple PHP array if it makes sense - something like:

```php
$iconmap = [
    /* ..... */
    'caret-down' => 'fa fa-caret-down',
    /* ..... */
];
```

Reply:
Again, I didn't want that logic in the blade, since there are other places where it could be useful, specifically when we're trying to determine which icon to use in transformers, even though we're not actually *using* that icon in those methods, we're passing it elsewhere (sometimes because it's in javascript, etc)

Reply:
fwiw, I'm for avoiding a giant `@php` block in the component too.